runtime/mainnet/src/config/revive.rs

```diff
@@ -0,0 +1,208 @@
+use frame_support::{
+	parameter_types,
+	traits::{ConstBool, ConstU32, ConstU64, Nothing},
+};
+use frame_system::EnsureSigned;
+
+use crate::{
+	deposit, Balance, Balances, Perbill, PolkadotXcm, Runtime, RuntimeCall, RuntimeEvent,
+	RuntimeHoldReason, Timestamp, TransactionPayment, UNIT,
+};
+
+// 18 decimals
+const ETH: u128 = 1_000_000_000_000_000_000;
+
+parameter_types! {
+	pub const DepositPerItem: Balance = deposit(1, 0);
+	pub const DepositPerByte: Balance = deposit(0, 1);
+	pub CodeHashLockupDepositPercent: Perbill = Perbill::from_percent(30);
+	pub const NativeToEthRatio: u32 = (ETH/UNIT) as u32;
+}
+
+impl pallet_revive::Config for Runtime {
+	type AddressMapper = pallet_revive::AccountId32Mapper<Self>;
+	// No runtime dispatchables are callable from contracts.
+	type CallFilter = Nothing;
+	type ChainExtension = ();
+	// EVM chain id. 3,395 is a unique ID still.
+	type ChainId = ConstU64<3_395>;
+	// 30 percent of storage deposit held for using a code hash.
+	type CodeHashLockupDepositPercent = CodeHashLockupDepositPercent;
+	type Currency = Balances;
+	type Debug = ();
+	type DepositPerByte = DepositPerByte;
+	type DepositPerItem = DepositPerItem;
+	type InstantiateOrigin = EnsureSigned<Self::AccountId>;
+	// 1 ETH : 1_000_000 UNIT
+	type NativeToEthRatio = NativeToEthRatio;
+	// 512 MB. Used in an integrity test that verifies the runtime has enough memory.
+	type PVFMemory = ConstU32<{ 512 * 1024 * 1024 }>;
+	type RuntimeCall = RuntimeCall;
+	type RuntimeEvent = RuntimeEvent;
+	type RuntimeHoldReason = RuntimeHoldReason;
+	// 128 MB. Used in an integrity that verifies the runtime has enough memory.
+	type RuntimeMemory = ConstU32<{ 128 * 1024 * 1024 }>;
+	type Time = Timestamp;
+	// Disables access to unsafe host fns such as xcm_send.
+	type UnsafeUnstableInterface = ConstBool<false>;
+	type UploadOrigin = EnsureSigned<Self::AccountId>;
+	type WeightInfo = pallet_revive::weights::SubstrateWeight<Self>;
+	type WeightPrice = TransactionPayment;
+	type Xcm = PolkadotXcm;
+}
+
+impl TryFrom<RuntimeCall> for pallet_revive::Call<Runtime> {
+	type Error = ();
+
+	fn try_from(value: RuntimeCall) -> Result<Self, Self::Error> {
+		match value {
+			RuntimeCall::Revive(call) => Ok(call),
+			_ => Err(()),
+		}
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use std::any::TypeId;
+
+	use frame_support::pallet_prelude::Get;
+	use pallet_revive::Config;
+
+	use super::*;
+
+	// 18 decimals
+	const ONE_ETH: u128 = 1_000_000_000_000_000_000;
+
+	#[test]
+	fn address_mapper_is_account_id32_mapper() {
+		assert_eq!(
+			TypeId::of::<<Runtime as Config>::AddressMapper>(),
+			TypeId::of::<pallet_revive::AccountId32Mapper<Runtime>>(),
+		);
+	}
+
+	#[test]
+	fn call_filter_is_nothing() {
+		assert_eq!(
+			TypeId::of::<<Runtime as pallet_revive::Config>::CallFilter>(),
```

Comment:
`pallet_revive::Config` is imported on line 70.

`Config` doesn't seem to be ambiguous in this context and is also used without the prefix in some other tests below.

Reply:
nice shout, thanks!

Reply:
https://github.com/r0gue-io/pop-node/pull/452/commits/4f24513d9514190226f55456a7bfed80bb207657